### PR TITLE
Builders and Implementations For Interaction Response

### PR DESCRIPTION
**Pull request by SinsofSloth**
Feel free to rename the methods and struct names if they're too wordy or redudant.

### CODE REVIEW

**Comment by Th3-M4jor** [comment]
This probably needs extra feature gates added if you want it to pass all the tests

**Comment by SinsofSloth** [comment]
Thanks for the heads-up! I was writing this with the feature enabled, so I didn't get these errors.

**Review by arqunis — APPROVED** [review]
